nx/lib/nx.ex

```diff
@@ -7046,6 +7046,152 @@ defmodule Nx do
     impl!(tensor).sort(tensor, tensor, axis: axis, comparator: comparator)
   end
 
+  @doc """
+  Calculates the QR decomposition of an 2-D tensor with shape `{M, N}` via the Householder method.
+
+  ## Options
+
+  `:mode` - Can be one of `:reduced`, `:complete`, `:r` or `:raw`. Defaults to `:reduced`
+    For the following, `K = min(M, N)`
+
+    * `:reduced` - returns `q` and `r` with shapes `{M, K}` and `{K, N}`
+    * `:complete` - returns `q` and `r` with shapes `{M, M}` and `{M, N}`
+    * `:r` - returns `r` with shape `{K, N}`
+    * `:raw` - returns `{h, tau}`, with `h` being the array of Householder reflectors and
```

Comment:
Almost forgot to ask: does anyone know what the `tau` parameter means?

Reply:
For simplicity, we can just omit `:r` and `:raw` modes and only return `:reduced` and `:complete`, wdyt? Note I'm referring to what the [jax.lax.linalg.qr](https://github.com/google/jax/blob/master/jax/_src/lax/linalg.py#L147) primitive does for this one!